core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -159,6 +159,23 @@ public void log(String format, Object... args) {
         }
     }
 
+    /**
+     * This will log {@link RuntimeException}, if logging is enabled , and throw the runtime exception.
+     * @param runtimeException to be thrown. It will do nothing if {@code null} is provided.
+     * @throws RuntimeException which is requested by this call.
+     */
+    public void logAndThrow(RuntimeException runtimeException) {
+        if (runtimeException == null) {
+            return;
+        }
+
+        // it will only log if log level is enabled in configuration
+        if (canLogAtLevel(level)) {
```

Comment:
This check won't always work as expected and won't comply with the changes made in PR #4194.

Reply:
@hemanttanwar Did you reconcile what this comment means? It would be good to see a follow-up comment to clarify we're on the same page.

Reply:
Spoke to @alzimmermsft and he is fine with the changes done to handle this case.